modules/apps/transfer/keeper/msg_server_test.go

```diff
@@ -7,6 +7,34 @@ import (
 	"github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
 )
 
+func checkEvents(
+	suite *KeeperTestSuite,
+	actualEvents sdk.Events,
+	expEvents map[string]map[string]string,
+) {
+	hasEvents := make(map[string]bool)
+	for eventType := range expEvents {
+		hasEvents[eventType] = false
+	}
+
+	for _, event := range actualEvents {
```

Comment:
I think I would prefer a `assertTransferEvents` function. Testing functions typically are not importable across packages, so I think it makes sense to create a function which understand the transfer event type and asserts its structure:

```go
func() assertTransferEvents(coin) {

expEvents := map[string]string{
	"sender":   suite.chainA.SenderAccount.GetAddress().String(),
	"receiver": suite.chainB.SenderAccount.GetAddress().String(),
	"amount":   coin.Amount.String(),
	"denom":    coin.Denom,
}

for _, event := range events {
    if event.Type == transfertypes.EventType {
        for _, attr := range event.Attributes {
            expValue, found := expEvent[string(attr.Key)]
	    suite.Require().True(found)
	    suite.Require().Equal(expValue, string(attr.Value))
        }
    }
}

Reply:
This will again get into the extra boilerplate problem that @crodriguezvega was mentioning. 
For this PR i dont mind having a function that is very specific, but for general events assertion, it might make more sense to take the event itself as a key, so we can assert all sorts of events.

As for importability is concerned, maybe it makes sense to have a `testutil` package like the sdk, and can have these bolier plate functions there.

Reply:
I made the change, but i am not in huge favour of this way of test functions, since it becomes less obvious in the test code flow, what is being asserted. Using an `expected` vs `actual` makes more intutive sense.
Also the `checkTransferEvents` function signature becomes a bit more ugly with more fields to it, example memo now.

Have a look at let me know if this is ok, or if i can revert it

Reply:
Happy to go with what you think works best. I would keep the `event.Type == transfertypes.EventTypeTransfer` since there's only one expected event type. The function naming I think can remain, it isn't accessible outside of the transfer testing package

Reply:
we can keep this for now then in the bigger issue, for event testing let me bring this up.
Thanks for your patience and discussion on this.

Reply:
Resolving the conversation, since addressed